src/amp_local_goal/amp_local_goal/follow_waypoint.py

```diff
@@ -0,0 +1,196 @@
+        costmapResolution = msg.metadata.resolution
+        xOffset = msg.metadata.origin.position.x
+        yOffset = msg.metadata.origin.position.y
+        costmapArray = np.zeros((costmapHeight, costmapWidth), dtype=np.uint8)
+        costmapIndex = 0
+        for i in range(costmapHeight):
+            for j in range(costmapWidth):
+                costmapArray[i][j] = COST_IMAGE_MAPPINGS.get(
+                    costmap[costmapIndex], costmap[costmapIndex])
+                costmapIndex += 1
+
+        # threshold for costmap values that meet our minimum (atm, just under inflation buffer)
+        threshCostmap = cv2.threshold(costmapArray, 100, 255,
+                                      cv2.THRESH_BINARY_INV)[1]
+        # add border around image to disincentivize picking values that are far from the kart
+        threshCostmap = cv2.copyMakeBorder(threshCostmap, 1, 1, 1, 1,
+                                           cv2.BORDER_CONSTANT, (0))
+
+        # apply parabola mask to image
+        mask = np.zeros((costmapHeight, costmapWidth))
+        mask_x_values = np.linspace(-1, 1, costmapWidth * 2)
+        mask_y_values = (self.parabola_numer / self.parabola_denom) * (
+            (mask_x_values)**2)
+        mask_x_indices = np.floor(
+            ((mask_x_values + 1) / 2) * (costmapWidth - 1)).astype(int)
+        mask_y_indices = np.floor(
+            ((mask_y_values + 1) / 2) * (costmapHeight - 1)).astype(int)
+        mask[mask_y_indices, mask_x_indices] = 1
+        for value in zip(mask_y_indices, mask_x_indices):
+            mask[0:value[0], value[1]] = 1
+        mask = scipy.ndimage.rotate(
+            mask,
+            angle=np.rad2deg(-1 *
+                             (self.currentKartYaw - self.initialKartYaw)) - 90,
+            reshape=False,
+            mode='nearest')
+        mask = cv2.copyMakeBorder(mask, 1, 1, 1, 1, cv2.BORDER_CONSTANT, (0))
+        maskedThershCostmap = (threshCostmap * mask).astype('uint8')
+
+        # Compute Distance Transform
+        distImg = cv2.distanceTransform(maskedThershCostmap, cv2.DIST_L2, 5)
+        distImgNormalized = distImg.copy()
+        cv2.normalize(distImg, distImgNormalized, 0, 1.0, cv2.NORM_MINMAX)
+
+        # Find point furthest away from all obstacles
+        max_loc = cv2.minMaxLoc(distImgNormalized)[3]
+
+        # scale points to real world distance
+        max_loc_costmap_origin = max_loc
+        max_loc_costmap_origin_scaled = tuple(
+            cord * costmapResolution for cord in max_loc_costmap_origin)
+
+        # move points to current offset of local map from global origin
+        max_loc_costmap_origin_offset = (max_loc_costmap_origin_scaled[0] +
+                                         xOffset,
+                                         max_loc_costmap_origin_scaled[1] +
+                                         yOffset)
+
+        # log values
+        self.get_logger().info(f"{[centx, centy]}")
```

Comment:
```suggestion
        self.get_logger().info(f"Goal pixel coordinates: {[max_loc[0], max_loc[1]]}")
```
Does `centx` exist is `show_costmaps` is False?